kotlin/perform-valve-isolation-trace/src/main/java/com/esri/arcgisruntime/sample/performvalveisolationtrace/MainActivity.kt

```diff
@@ -0,0 +1,296 @@
+import com.esri.arcgisruntime.utilitynetworks.UtilityCategory
+import com.esri.arcgisruntime.utilitynetworks.UtilityCategoryComparison
+import com.esri.arcgisruntime.utilitynetworks.UtilityCategoryComparisonOperator
+import com.esri.arcgisruntime.utilitynetworks.UtilityElement
+import com.esri.arcgisruntime.utilitynetworks.UtilityElementTraceResult
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetwork
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceConfiguration
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceFilter
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceParameters
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceType
+import kotlinx.android.synthetic.main.activity_main.*
+import java.util.UUID
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG = MainActivity::class.java.simpleName
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a basemap and set it to the map view
+    val map = ArcGISMap(Basemap.createStreetsNightVector())
+
+    // load the utility network data from the feature service and create feature layers
+    val distributionLineFeatureTable =
+      ServiceFeatureTable(getString(R.string.distribution_line_url))
+    val distributionLineLayer = FeatureLayer(distributionLineFeatureTable)
+    val deviceFeatureTable =
+      ServiceFeatureTable(getString(R.string.device_url))
+    val deviceLayer = FeatureLayer(deviceFeatureTable)
+
+    // add the feature layers to the map
+    map.operationalLayers.addAll(arrayOf(distributionLineLayer, deviceLayer))
+
+    mapView.map = map
+
+    // make sure the fab doesn't obscure the attribution bar
+    mapView.addAttributionViewLayoutChangeListener { _, _, _, _, bottom, _, _, _, oldBottom ->
+      val layoutParams = fab.layoutParams as CoordinatorLayout.LayoutParams
+      layoutParams.bottomMargin += bottom - oldBottom
+    }
+
+    // show the options sheet when the floating action button is clicked
+    fab.setOnClickListener {
+      fab.isExpanded = !fab.isExpanded
+    }
+
+    // close the options sheet when the map is tapped
+    mapView.onTouchListener = object : DefaultMapViewOnTouchListener(this, mapView) {
+      override fun onTouch(view: View?, event: MotionEvent?): Boolean {
+        if (fab.isExpanded) {
+          fab.isExpanded = false
+        }
+        return super.onTouch(view, event)
+      }
+    }
+
+    // create and load the utility network
+    map.addDoneLoadingListener {
+      createUtilityNetwork()
+    }
+  }
+
+  /**
+   * Creates a utility network from the string resource url and loads it.
+   * Initializes the starting point and UI.
+   */
+  private fun createUtilityNetwork(
+  ) {
+    // create a graphics overlay for the starting location and add it to the map view
+    val startingLocationGraphicsOverlay = GraphicsOverlay().also {
```

Comment:
this could probably be instantiated by lazy

Reply:
Should it though? I thought it was preferable to keep values in the scope of the function if that's the only place it's used

Reply:
I'd suggest creating the graphics overlay as lazy

and doing mapView.graphicsOverlays.add in your mapView apply block above

Reply:
I've changed it, but it felt out of place to be creating the symbol etc. in that function, so I moved that to onCreate(), but it turns out the starting location symbol needs to be in the class scope as well for that to work. So now the starting location symbol is being instantiated by lazy as well.